src/cpp/astar.cpp

```diff
@@ -73,6 +96,29 @@ static PyObject *astar(PyObject *self, PyObject *args) {
   nodes_to_visit.push(start_node);
 
   int* nbrs = new int[8];
+  
+  // Get the heuristic method to use
+  float (*heuristic_func)(int, int, int, int, int, int);
+  
+  if (heuristic_override == 1) {
+    heuristic_func = linf_norm;
+  } else if (heuristic_override == 2) {
+    heuristic_func = l1_norm;
+  } else if (heuristic_override == 3) {
+    heuristic_func = l1_orthogonal_x;
+  } else if (heuristic_override == 4) {
+    heuristic_func = l1_orthogonal_y;
```

Comment:
What do you think about separating the non-default cases into their own dispatch function?
```C++
if (heuristic_override == DEFAULT) {
  // do the default stuff
}
else {
  heuristic_func = select_heuristic(heuristic_override)
}
```

Reply:
I think pretty easy to understand at the moment

Reply:
Done